plugins/pluginmanagers/lazy.nix

```
@@ -1,222 +1,204 @@
+          submodule (
+            { config, ... }:
+
+            let
+              cfg = config;
+            in
+            {
+              freeformType = attrsOf anything;
+              options = {
+                __unkeyed = mkNullOrOption str ''
+                  The "unkeyed" attribute is the plugin's short plugin url.
+                  It ill be expanded using `config.git.url_format`. It can
+                  also be a url or dir.
+                '';
+
+                dir = mkNullOrOption str "A directory pointing to a local plugin";
+
+                url = mkNullOrOption str "A custom git url where the plugin is hosted";
+
+                pkg = mkNullOrOption package "Vim plugin to install";
+
+                name = mkNullOrOption str "Name of the plugin to install";
+
+                dev = defaultNullOpts.mkBool false ''
+                  When true, a local plugin directory will be used instead.
+                  See config.dev
+                '';
+
+                lazy = defaultNullOpts.mkBool true ''
+                  When true, the plugin will only be loaded when needed.
+                  Lazy-loaded plugins are automatically loaded when their Lua modules are required,
+                  or when one of the lazy-loading handlers triggers
+                '';
+
+                enabled = defaultNullOpts.mkStrLuaFnOr bool "`true`" ''
+                  When false then this plugin will not be included in the spec. (accepts fun():boolean)
+                '';
+
+                cond = defaultNullOpts.mkStrLuaFnOr bool "`true`" ''
+                  When false, or if the function returns false,
+                  then this plugin will not be loaded. Useful to disable some plugins in vscode,
+                  or firenvim for example. (accepts fun(LazyPlugin):boolean)
+                '';
+
+                # WARNING: Be very careful if changing the type of
+                # `dependencies`. Choosing the wrong type may cause a stack
+                # overflow due to infinite recursion, and it's very possible
+                # that the test cases won't catch this problem. To be safe,
+                # perform thorough manual testing if you do change the type of
+                # `dependencies`. Also use `types.eitherRecursive` instead of
+                # `types.either` here, as using just `types.either` also leads
+                # to stack overflow. 
+                dependencies = mkNullOrOption (eitherRecursive lazyPluginType lazyPluginsListType) ''
+                  A list of plugin names or plugin specs that should be
+                  loaded when the plugin loads. Dependencies are always
+                  lazy-loaded unless specified otherwise. When specifying a
+                  name, make sure the plugin spec has been defined somewhere
+                  else. This can also be a single string such as for a short
+                    plugin url (See: https://lazy.folke.io/spec).
+                '';
```

```suggestion
                dependencies = mkNullOrOption (eitherRecursive lazyPluginType lazyPluginsListType) ''
                  A list of plugin names or plugin specs that should be
                  loaded when the plugin loads.
                  
                  Dependencies are always lazy-loaded unless specified otherwise.
                  
                  When specifying a name, make sure the plugin spec has been defined somewhere else.
                  This can also be a single string such as for a short plugin URL. (See: https://lazy.folke.io/spec).
                '';
```

Should this text be broken up more and maybe give examples?